extensions/interactions/MathEquationInput/MathEquationInput_test.py

```diff
@@ -0,0 +1,187 @@
+
+    def test_math_equation_input_converted_to_proto_correctly(self):
+        interaction_dict = {
+            'id': 'MathEquationInput',
+            'customization_args': {
+                'customOskLetters': {
+                     'value': ['\u03C0', '\u03C0']
+                },
+                'useFractionForDivision': {
+                    'value': False
+                }
+            },
+            'answer_groups': [{
+                'outcome': {
+                    'dest': 'Number With Units',
+                    'feedback': {
+                        'content_id': 'feedback_1',
+                        'html': '<p>Yes, well done!</p>'
+                    },
+                    'labelled_as_correct': False,
+                    'param_changes': [],
+                    'refresher_exploration_id': None,
+                    'missing_prerequisite_skill_id': None
+                },
+                'rule_specs': [{
+                    'rule_type': 'MatchesExactlyWith',
+                    'inputs': {
+                        'x': 'a+b=c',
+                        'y': 'both'
+                    }
+                }, {
+                    'rule_type': 'IsEquivalentTo',
+                    'inputs': {
+                        'x': 'a+b=c'
+                    }
+                }, {
+                    'rule_type': 'MatchesUpToTrivialManipulations',
+                    'inputs': {
+                        'x': 'a+b=c'
+                    }
+                }],
+                'training_data': [],
+                'tagged_skill_misconception_id': 'skill_id-misconception_id'
+            }],
+            'default_outcome': {
+                'dest': 'Introduction',
+                'feedback': {
+                    'content_id': 'default_outcome',
+                    'html': '<p> introduce </p>'
+                },
+                'labelled_as_correct': False,
+                'param_changes': [],
+                'refresher_exploration_id': None,
+                'missing_prerequisite_skill_id': None
+            },
+            'confirmed_unclassified_answers': [],
+            'hints': [{
+                'hint_content': {
+                    'content_id': 'hint_1',
+                    'html': '<p>This is a first hint.</p>'
+                }
+            }, {
+                'hint_content': {
+                    'content_id': 'hint_2',
+                    'html': '<p>This is the second hint.</p>'
+                }
+            }],
+            'solution': {
+                'answer_is_exclusive': True,
+                'correct_answer': 'a+b=c',
+                'explanation': {
+                    'content_id': 'solution',
+                    'html': '<p>This is solution for algebric.</p>'
+                }
+            }
+        }
+        registery = interaction_registry.Registry
+        math_instance = registery.get_interaction_by_id('MathEquationInput')
```

Comment:
Use math_equation rather than math, throughout (ditto below).

Also perhaps pick a clearer name for this -- "instance" is vague. What is the object being referenced, exactly -- what noun describes it accurately?

Reply:
I had updated these with the name of interactions only, as `get_interaction_by_id ` will return an interaction. 
Updated in all the respect test files in this PR.